src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/SyntaxExtensions.cs

```diff
@@ -81,6 +81,20 @@ public static SyntaxTokenList StripTriviaFromTokens(this SyntaxTokenList tokenLi
             return new SyntaxTokenList(strippedTokens);
         }
 
+        public static SyntaxTokenList StripAccessibilityModifiers(this SyntaxTokenList tokenList)
+        {
+            SyntaxToken[] strippedTokens = new SyntaxToken[tokenList.Count];
+            for (int i = 0; i < tokenList.Count; i++)
+            {
+                if (tokenList[i].Kind() is SyntaxKind.PublicKeyword or SyntaxKind.InternalKeyword or SyntaxKind.ProtectedKeyword or SyntaxKind.PrivateKeyword)
+                {
+                    continue;
+                }
+                strippedTokens[i] = tokenList[i];
+            }
+            return new SyntaxTokenList(strippedTokens);
```

Comment:
nit: I'd rather do a `List<SytnaxToken>` here than an array so we don't have empty tokens we're carrying around.